src/clout/core.cljc

```diff
@@ -2,24 +2,39 @@
   "A small language for routing."
   (:require [clojure.string :as string]
             [clojure.set :as set]
-            [instaparse.core :as insta]))
+            [instaparse.core :as insta]
+            #?(:cljs [clout.regex :refer [pL]])))
 
 (def ^:private re-chars (set "\\.*+|?()[]{}$^"))
+#?(:cljs (def re-char-keys (seq re-chars)))
+#?(:cljs (def re-char-vals (map #(str "\\" %) re-char-keys)))
 
-(defn- re-escape [s]
-  (string/escape s #(if (re-chars %) (str \\ %))))
+#?(:clj (def ^:private pL "\\p{L}"))
 
-(defn- re-groups* [^java.util.regex.Matcher matcher]
-  (for [i (range (.groupCount matcher))]
-    (.group matcher (int (inc i)))))
+(defn- re-escape [s]
+  #?(:clj
+     (string/escape s #(if (re-chars %) (str \\ %)))
+     :cljs
+     (string/escape s (zipmap re-char-keys re-char-vals))))
+
+#?(:clj
+   (defn- re-groups*
+     [^java.util.regex.Matcher matcher]
+     (for [i (range (.groupCount matcher))]
+       (.group matcher (int (inc i))))))
+
+#?(:cljs
+   (defn re-groups**
+     [re s]
+     (-> (re-seq re s) flatten rest)))
```

Comment:
Can you explain how this is equivalent to `re-groups*`?

Reply:
For example re = `(\d{3})-(\d{3})-(\d{3})`, s = `111-222-333`
```
=> (re-seq re s)
(["111-222-333" "111" "222" "333"])
=> (flatten (re-seq re s))
("111-222-333" "111" "222" "333")
=> (rest (flatten (re-seq re s)))
("111" "222" "333")
```

Reply:
I don't think this is correct. You're using `re-seq` in the ClojureScript version (matching zero or more patterns), whereas the Clojure version uses the equivalent of `re-matches` (matching exactly one pattern).

What we probably want is something more like:

```clojure
(defn- re-matches* [re s]
  #?(:clj
     (let [matcher (re-matcher re s)]
       (when (.matches matcher)
         (for [i (range (.groupCount matcher))]
           (.group matcher (int (inc i))))))
     :cljs
     (if (string? s)
       (let [matches (.exec re s)]
         (when (= (first matches) s)
           (seq matches))))
       (throw (js/TypeError. "re-matches* must match against a string.")))))
```

And do away with `re-groups*`